tests/framework/common.go

```diff
@@ -540,12 +540,23 @@ func (td *OsmTestData) RestartOSMController(instOpts InstallOSMOpts) error {
 		LabelSelector: labels.Set(labelSelector.MatchLabels).String(),
 	}
 
+	getOptions := metav1.GetOptions{}
+
 	controllerPods, err := td.Client.CoreV1().Pods(instOpts.ControlPlaneNS).List(context.TODO(), listOptions)
+	controllerDeployment, errSpec := td.Client.AppsV1().Deployments(instOpts.ControlPlaneNS).Get(context.TODO(), "osm-controller", getOptions)
+
 	if err != nil {
 		return errors.Wrap(err, "error fetching controller pod")
 	}
-	if len(controllerPods.Items) != 1 {
-		return errors.Errorf("expected 1 osm-controller pod, got %d", len(controllerPods.Items))
+
+	if errSpec != nil {
+		return errors.Wrap(err, "error fetching controller deployment")
```

Comment:
Please move the error checks right after the function calls, instead of consolidating them like this.